locks/dynamodb/dynamo_lock.go

```diff
@@ -86,8 +87,19 @@ func (dynamoDbLock DynamoDbLock) ReleaseLock(terragruntOptions *options.Terragru
 		return err
 	}
 
-	if err := removeItemFromLockTable(dynamoDbLock.StateFileId, dynamoDbLock.TableName, client); err != nil {
-		return err
+	err = removeItemFromLockTable(dynamoDbLock.StateFileId, dynamoDbLock.TableName, client)
+
+	// If the table or lock doesn't exist log it, else return the error
+	if err != nil {
+		if awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error); isAwsErr && awsErr.Code() == "ResourceNotFoundException" {
+			terragruntOptions.Logger.Printf("Table %s does not exist in DynamoDB!", dynamoDbLock.TableName)
+			return nil
+		} else if awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error); isAwsErr && awsErr.Code() == "ConditionalCheckFailedException" {
+			terragruntOptions.Logger.Printf("Nothing to release, specified lock does not exist in table %s", dynamoDbLock.TableName)
+			return nil
+		} else {
+			return err
+		}
```

Comment:
Could you extract this error-handling code within the `if err != nil` check into a separate method? The method could be called something like `handleRemoveItemFromLockTableError`.

Also, it seems strange to return `nil` in the case of recognized AWS errors. Why not wrap it with a clear error message? For example, you could declare:

```go
type TableDoesNotExist struct {
    TableName string
    Underlying  error
}
func (err TableDoesNotExist) Error() string {
  return fmt.Sprintf("Table %s does not exist in DynamoDB! Original error from AWS: %v", err.TableName, err.Underlying)
}
```

You could then return `TableDoesNotExist{dynamoDbLock.TableName, awsErr}` from the `handleRemoveItemFromLockTableError` method, and a nice, clean error message will be shown for you automatically rather than having to log it separately.